lib/room.h

```diff
@@ -508,6 +508,19 @@ class Room : public QObject {
     Q_INVOKABLE const Quotient::StateEventBase*
     getCurrentState(const QString& evtType, const QString& stateKey = {}) const;
 
+    /// Get all state events in the room.
+    /*! This method returns all known state events that have occured in
+     * the room, as a mapping from the event type and state key to value.
+     */
+    Q_INVOKABLE const QHash<StateEventKey, const StateEventBase*> stateEvents() const;
+
+    /// Get all state events in the room of a certain type.
+    /*! This method returns all known state events that have occured in
+     * the room of the given type.
+     */
+    Q_INVOKABLE const QList<const StateEventBase*>*
```

Comment:
Unlike the above, `Q_INVOKABLE` is fine here, with `QVector` being converted to a JavaScript array on the QML side (see https://doc.qt.io/qt-5/qtqml-cppintegration-data.html#sequence-type-to-javascript-array) thanks to `const StateEventBase*` being marked with `Q_DECLARE_METATYPE`.
```suggestion
    Q_INVOKABLE QVector<const StateEventBase*>
```